lib/public_key/src/pubkey_cert.erl

```diff
@@ -1149,45 +1186,1151 @@ is_double_quote("\"") ->
+    lists:map(
+      fun(#policy_tree_node{valid_policy = ValidPolicy, expected_policy_set = Set} = Node)
+            when ValidPolicy == IPolicy ->
+              case Set == [ValidPolicy] of
+                   true ->
+                       Node#policy_tree_node{expected_policy_set = [SPolicy]};
+                   false ->
+                      Node#policy_tree_node{expected_policy_set = Set ++ [SPolicy]}
+               end;
+         (Node) ->
+              Node
+      end, Nodes).
+
+is_policy_present(_, _, []) ->
+    false;
+is_policy_present(IPolicy, KeyPos, [Nodes | Rest]) ->
+    case lists:keymember(IPolicy, KeyPos, Nodes) of
+        true ->
+            true;
+        false ->
+            is_policy_present(IPolicy, KeyPos, Rest)
+    end.
+
+map_any_policy(IPolicy, SPolicy, AnyQualifiers, Nodes) ->
+    case lists:keyfind(?anyPolicy, #policy_tree_node.valid_policy, Nodes) of
+        #policy_tree_node{criticality_indicator = Critical} ->
+            [policy_node(Critical, IPolicy, AnyQualifiers, [SPolicy]) | Nodes];
+        false ->
+            Nodes
+    end.
+
+%% 6.1.4. i: If a policy constraints extension is included in the
+%% certificate, modify the explicit_policy and policy_mapping state
+%% variables as follows:
+
+%% (1) If requireExplicitPolicy is present and is less than
+%% explicit_policy, set explicit_policy to the value of
+%% requireExplicitPolicy.
+
+%% (2) If inhibitPolicyMapping is present and is less than
+%% policy_mapping, set policy_mapping to the value of
+%% inhibitPolicyMapping.
+handle_policy_constraints(#path_validation_state{
+                             policy_constraint_ext =
+                                 #'Extension'{extnID = ?'id-ce-policyConstraints',
+                                              extnValue =
+                                                  #'PolicyConstraints'{requireExplicitPolicy =
+                                                                           ExplicitPolicy,
+                                                                       inhibitPolicyMapping =
+                                                                           InhibitMapPolicy}},
+                             explicit_policy = CurrentExplitPolicyConstraint,
+                             inhibit_policy_mapping = CurrentPolicyMappingConstraint} =
+                              ValidationState) ->
+    ExplicitPolicyConstraint = policy_constraint(CurrentExplitPolicyConstraint, ExplicitPolicy),%6.5.1 b
+    PolicyMappingConstraint = policy_constraint(CurrentPolicyMappingConstraint, InhibitMapPolicy),
+    
+    handle_policy_constraints(
+      ValidationState#path_validation_state{explicit_policy = ExplicitPolicyConstraint,
+                                            inhibit_policy_mapping = PolicyMappingConstraint ,
+                                            policy_constraint_ext = undefined});
+
+handle_policy_constraints(
+  #path_validation_state{policy_constraint_ext =
```

Comment:
should this match towards policy_inhibit_ext?

Reply:
No same as below

Reply:
inhibitAnyPolicy extension is matched out from policy_constraint_ext whereas I think it should be stored in policy_inhibit_ext

Reply:
Yes agree there is something fish here, I am looking into it!

Reply:
This was a bug! Made a new fixup commit to be squashed later!